storage_service/locations/models/package.py

```diff
@@ -674,49 +674,81 @@ def _store_aip_to_pending(self, origin_location, origin_path):
             already_generated_ptr_exists=already_generated_ptr_exists)
 
     def _store_aip_to_uploaded(self, v, related_package_uuid):
-        """Get this AIP to the "uploaded" stage of ``store_aip`` by
-        1. moving it to the SS internal location,
-        2. setting its status to "staging",
-        3. calling ``post_move_to_storage_service`` on the source space,
-        4. moving it to the destination space/location,
-        5. setting the status to "uploaded" (if applicable),
-        6. setting a related package (if applicable),
-        7. calling ``post_move_from_storage_service`` on the destination space,
-        8. updating quotas on the destination space, and
-        9. persisting the package to the database.
+        """Get this AIP to the "uploaded" stage of ``store_aip``
         :param namedtuple v: object with attributes needed for processing.
         :param str related_package_uuid: UUID of a related package.
         :returns NoneType None:
         """
-        v.src_space.move_to_storage_service(
-            source_path=os.path.join(self.origin_location.relative_path,
-                                     self.origin_path),
-            destination_path=self.current_path,  # This should include Location.path
-            destination_space=v.dest_space)
-        self.status = Package.STAGING
-        self.save()
-        v.src_space.post_move_to_storage_service()
-        storage_effects = v.dest_space.move_from_storage_service(
-            source_path=self.current_path,  # This should include Location.path
-            destination_path=os.path.join(
+        if Space.posix_move_supported(v.src_space, v.dest_space):
```

Comment:
As detailed more below, I wonder if you could not instead refactor this to try the `posix_move` call and fallback to the `move_to/from` call if a `TypeError` is raised, which would remove the double call to `posix_move_supported`.